src/main/java/com/example/petbuddybackend/service/chat/ChatService.java

```diff
@@ -146,6 +147,44 @@ public MessageCallback createCallbackMessageSeen(Long chatId, String usernameToS
         };
     }
 
+    @Transactional(readOnly = true)
+    public ChatRoomDTO getChatRoomWithParticipant(
+            String username,
+            Role userRole,
+            String participantUsername,
+            ZoneId timeZone
+    ) {
+        ChatRoom chatRoom;
+        AppUser chatter;
+
+        switch(userRole) {
+            case CLIENT:
+                chatRoom = getByClientEmailAndCaretakerEmail(username, participantUsername);
+                chatter = chatRoom.getCaretaker().getAccountData();
+                break;
+            case CARETAKER:
+                chatRoom = getByClientEmailAndCaretakerEmail(participantUsername, username);
+                chatter = chatRoom.getClient().getAccountData();
+                break;
+            default:
+                throw new UnsupportedOperationException(String.format(INVALID_USER_ROLE_MESSAGE, userRole));
+        }
+
+        ChatMessage lastMessage = chatMessageRepository.findFirstByChatRoom_IdOrderByCreatedAtDesc(chatRoom.getId());
+        boolean isSeenByPrincipal = messageSeenByPrincipal(username, lastMessage);
+        return chatMapper.mapToChatRoomDTO(chatRoom.getId(), chatter, lastMessage, isSeenByPrincipal, timeZone);
+    }
+
+    private boolean messageSeenByPrincipal(String username, ChatMessage lastMessage) {
+        String senderEmail = lastMessage.getSender().getEmail();
+
+        if(senderEmail.equals(username)) {
+            return true;
+        }
+
+        return lastMessage.getSeenByRecipient();
```

Comment:
a jezeli username jest senderem to nie wystarczy i tak zrobic: 
return lastMessage.getSeenByRecipient();?

Reply:
Recipient to jest odbiorca. Funkcja zwraca seenByPrincipal. Jeżeli username jest senderem to oznacza, że na pewno odczytał wiadomość i stąd ten `return true`

Reply:
A ok, a jeśli jest odbiorca to wtedy w zależności od tego co zwróci get